### PR TITLE
feat: Add trust assumptions, new functionality to README (SC-863)

### PR DESCRIPTION
**Pull request by lucas-manuel**
Addresses https://github.com/cantinasec/maker-review-1205/issues/9

### CODE REVIEW

**Comment by lucas-manuel** [comment]
Addresses https://github.com/cantinasec/maker-review-1205/issues/9

File: README.md [hunk 1]
```diff
@@ -68,6 +73,30 @@ The rate limit is calculated as follows:
 
 This is a linear rate limit that increases over time with a maximum limit. This rate limit is derived from these values which can be set by and admin OR updated by the `CONTROLLER` role. The `CONTROLLER` updates these values to increase/decrease the rate limit based on the functionality within the contract (e.g., decrease the rate limit after minting USDS by the minted amount by decrementing `lastAmount` and setting `lastUpdated` to `block.timestamp`).
 
+## Trust Assumptions
+Below are all stated trust assumptions for using this contract in production:
+- The `DEFAULT_ADMIN_ROLE` is fully trusted, to be run by governance.
+- The `RELAYER` role is assumed to be able to be fully compromised by a malicious actor. 
+  - The logic in the smart contracts must prevent the movement of value anywhere outside of the ALM system of contracts.
+  - The rate limits setup must prevent the possibility of griefing or locking/burning value by a malicious actor.
+  - The `FREEZER` must be able to stop the compromised `RELAYER` from performing more harmful actions by using the `freeze()` function.
```
Comment (hexonaut) [inline README.md:82]: I would rephrase these - they are pretty vague.

```
  - Any action must be limited to "reasonable" slippage/losses/opportunity cost by rate limits. A malicious relayer should not be able to cause loses above 50bps of the principal in a 24 hours time frame.
  - The `FREEZER` must be able to stop the compromised `RELAYER` from performing more harmful actions to within the max rate limits by using the `freeze()` function.
```

File: README.md [hunk 2]
```diff
@@ -68,6 +73,30 @@ The rate limit is calculated as follows:
...
+- AAVE deposits are subject to AAVE governance changes, such as setting reserves inactive, paused or frozen. 
+- USDC holdings are subject to blacklist risk.
+- Ethena deposits are subject to USDe depeg risks, blacklist risks, and mint and burn limits.
+- The Ethena Minter is fully trusted. If it is compromised it can: 
+  - Accept orders submitted by a malicious delegated signer that would burn a large amount of USDC for a small amount of USDe (bad quote).
+  - DOS minting/burning of USDe, locking funds.
+- A compromised `RELAYER` can DOS Ethena unstaking, but this can be mitigated by freezing the Controller and reassigning the `RELAYER`. This is outlined in a test `test_compromisedRelayer_lockingFundsInEthenaSilo`.
```
Comment (hexonaut) [inline README.md:89]: I don't think we need to exhaustively list every risk lol. There is an uncountable amount. You've listed USDe having depeg risk, but every derivative has this.

Even just the trust assumptions are incredibly complex for something like Ethena and is best left to risk analysis of onboarding process and not listed in this readme.

File: README.md [hunk 3]
```diff
@@ -68,6 +73,30 @@ The rate limit is calculated as follows:
...
+
+## Operational Requirements
+- All ERC-4626 vaults that are onboarded MUST have an initial burned shares amount that prevents rounding-based frontrunning attacks. These shares have to be unrecoverable so that they cannot be removed at a later date.
+- All ERC-20 tokens are to be non-rebasing with sufficiently high decimal precision.
+- Rate limits must be configured for specific ERC-4626 vaults and AAVE aTokens (vaults without rate limits set will revert). Unlimited rate limits can be used as an onboarding tool.
+- Rate limits must take into account:
+  - Risk tolerance for a given protocol
+  - Griefing attacks (e.g., repetitive transactions with high slippage by malicious relayer).
+  - Reputational attack vectors (e.g., sudden withdraws from utilization-based markets spiking interest rates).
```
Comment (hexonaut) [inline README.md:98]: This is less of a concern. The withdrawal issue is more if there is a delay. For example, being able to withdraw 1b sUSDe all at once will not cause slippage or even reputational damage, but it will carry a huge opportunity cost. Spiking interest rates is less of a concern for the purposes of this.

**Review by hexonaut — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (oldchili) [reply README.md:89]: Agree, I think it would be good to mention that these are examples, so it would not be regarded as a great deal if someone comes up with something that is not listed here.

Comment (hexonaut) [reply README.md:89]: I don't think they are relevant at all for a technical audit. This is the type of analysis that is done by collateral onboarding.

The security model is that when we ship assets to protocol x, we trust whatever assumptions have been made by the collateral onboarding process.

Where the audits should focus is to make sure that whatever the swap mechanism is, there is limited damage the `relayer` can do in terms of slippage and opportunity cost. The trust model assumes that wherever we ship the assets after that the protocol is completely trusted.

File: README.md [hunk 4]
```diff
@@ -68,6 +73,23 @@ The rate limit is calculated as follows:
 
 This is a linear rate limit that increases over time with a maximum limit. This rate limit is derived from these values which can be set by and admin OR updated by the `CONTROLLER` role. The `CONTROLLER` updates these values to increase/decrease the rate limit based on the functionality within the contract (e.g., decrease the rate limit after minting USDS by the minted amount by decrementing `lastAmount` and setting `lastUpdated` to `block.timestamp`).
 
+## Trust Assumptions and Attack Mitigation
+Below are all stated trust assumptions for using this contract in production:
+- The `DEFAULT_ADMIN_ROLE` is fully trusted, to be run by governance.
+- The `RELAYER` role is assumed to be able to be fully compromised by a malicious actor. **This should be a major consideration during auditing engagements.**
+  - The logic in the smart contracts must prevent the movement of value anywhere outside of the ALM system of contracts.
+  - Any action must be limited to "reasonable" slippage/losses/opportunity cost by rate limits. **A malicious relayer should not be able to cause losses above 50bps of the principal in a 24 hours time frame.**
```
Comment (hexonaut) [inline README.md:81]: Remove `malicious relayer should not be able to cause losses above 50bps of the principal in a 24 hours time frame`

I wrote that originally, but not sure if it makes much sense in the abstract. I think having the more open “reasonable” losses makes sense so we potentially get false positives vs missing something.
